llvm/lib/SYCLLowerIR/SYCLAddOptLevelAttribute.cpp

```diff
@@ -0,0 +1,28 @@
+//===---- SYCLAddOptLevelAttribute.cpp - SYCLAddOptLevelAttribute Pass ---===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===---------------------------------------------------------------------===//
+//
+// Pass adds 'sycl-optlevel' function attribute based on optimization level
+// passed in.
+//===---------------------------------------------------------------------===//
+
+#include "llvm/SYCLLowerIR/SYCLAddOptLevelAttribute.h"
+
+#include "llvm/IR/Module.h"
+
+using namespace llvm;
+
+PreservedAnalyses
+SYCLAddOptLevelAttributePass::run(Module &M, ModuleAnalysisManager &MAM) {
+  for (Function &F : M.functions()) {
+    // Here, we add a function attribute 'sycl-optlevel' to store the
```

Comment:
We can add a filter here to skip declarations, at least

Reply:
That's a good idea. Thanks